File: backend/mallapi/src/main/java/org/nareun130/mallapi/dto/ProductDTO.java
package org.nareun130.mallapi.dto;

import java.util.ArrayList;
import java.util.List;

import org.springframework.web.multipart.MultipartFile;

import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class ProductDTO {
    
    private Long pno;

    private String pname;

    private int price;

    private String pdesc;

    private boolean delFlag;

    @Size(min = 1, message = "이미지를 1개 이상 첨부하여야 합니다.")
    @Builder.Default
    private List<MultipartFile> files = new ArrayList<>();

    @Builder.Default
    private List<String> uploadFileNames = new ArrayList<>();
}

File: backend/mallapi-kotlin/src/main/kotlin/com/nareun/mallapikotlin/dto/ProductDTO.kt
package com.nareun.mallapikotlin.dto

import org.springframework.web.multipart.MultipartFile

data class ProductDTO (
    var pno :Long=0,
    var pname : String="",
    var price: Int = 0,
    var pdesc : String = "",
    var deFlag : Boolean = false,
    //실제 파일 데이터
    var files: List<MultipartFile> ?= mutableListOf(),
    // 업로드된 파일 결과들
    var uploadFileNames:List<String> ?= mutableListOf()

){
    //ModelMapper를 위한 기본 생성자
    constructor(): this(0,"",0,"",false, mutableListOf(), mutableListOf())
}